src/functions.php

```diff
@@ -75,22 +77,37 @@ function ($error) use (&$exception, &$rejected, &$wait) {
     // argument does not show up in the stack trace in PHP 7+ only.
     $promise = null;
 
+    if ($rejected) {
+        // promise is rejected with an unexpected value (Promise API v1 or v2 only)
+        if (!$rejectedThrowable instanceof \Exception && !$rejectedThrowable instanceof \Throwable) {
+            $rejectedThrowable = new \UnexpectedValueException(
+                'Promise rejected with unexpected value of type ' . (is_object($rejectedThrowable) ? get_class($rejectedThrowable) : gettype($rejectedThrowable))
+            );
+        }
+
+        throw $rejectedThrowable;
+    }
```

Comment:
@WyriHaximus I can see why this was done and even though I'm not a big fan of the duplication, I would be happy to accept this as is for the time being (continuous improvement). Unfortunately, it looks like this currently reduces code coverage from 100% to ~97%. Can you look into this? (Same as in #21)

Reply:
Will toss it in a function, not really a fan of either solution tbh. (Same as in #21)

Reply:
@clue Pushed the deduplication (Same as #21)

Reply:
@clue Works for me. But I've seen that code around a lot lately. Would it make sense to, in time, put it in a helper/utility package?

Reply:
@clue Working on the test failures, there is an interesting flaw that causes an infinite loop

Reply:
Ok fixed it :sunglasses: